File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
from collections import Counter

def solution(s):
    answer = True
    
    counter = Counter(s)
    stack = []
    
    if counter['('] != counter[')']:
        return False
    
    for i  in s:
        if i == ')' and stack and stack[-1] == '(':
            stack.pop()
        elif i == '(':
            stack.append(i)
        
    if len(stack) > 0 :
        return False
    
    return True

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s) {
  const stack = [];
    
  for (let ch of s) {
    if (ch === '(') {
      stack.push(ch);
    } 
    else {
      if (stack.length === 0) return false;
      stack.pop();
    }
  }
  return stack.length === 0;
}
